packages/cozy-codemods/src/transforms/transform-radio.js

```diff
@@ -0,0 +1,132 @@
+      const parentAttributeName =
+        nodePath?.parentPath?.parentPath?.value?.name?.name
+      const isInCompWithControlAttribute = parentAttributeName === 'control'
+
+      if (isInCompWithControlAttribute) {
+        return nodePath.value
+      }
+
+      const { attributes } = nodePath.node.openingElement
+      const valueAtttribute = getAttribute(attributes, 'value')
+      const labelAttribute = getAttribute(attributes, 'label')
+
+      const formControlLabelAttributes = [
+        j.jsxAttribute(
+          j.jsxIdentifier('control'),
+          j.jsxExpressionContainer(nodePath.value)
+        )
+      ]
+      if (valueAtttribute) {
+        formControlLabelAttributes.push(
+          j.jsxAttribute(j.jsxIdentifier('value'), valueAtttribute.value)
+        )
+      }
+      if (labelAttribute) {
+        formControlLabelAttributes.push(
+          j.jsxAttribute(j.jsxIdentifier('label'), labelAttribute.value)
+        )
+      }
+
+      // create FormControlLabel component with Radio in its control prop
+      const jsxFormControlLabel = j.jsxElement(
+        j.jsxOpeningElement(
+          j.jsxIdentifier('FormControlLabel'),
+          formControlLabelAttributes.reverse()
+        )
+      )
+      jsxFormControlLabel.selfClosing = true
+      jsxFormControlLabel.openingElement.selfClosing = true
+
+      // remove useless attribute from Radio
+      remove(attributes, attribute => getAttributeName(attribute) === 'value')
+      remove(attributes, attribute => getAttributeName(attribute) === 'label')
+
+      // wrap Radio in FormControlLabel
+      return jsxFormControlLabel
+    })
+  }
+
+  // Radio transformation
+  const transformRadio = nodePath => {
+    const { attributes } = nodePath.node.openingElement
+
+    // add edge prop if previous Radio had no gutter prop
+    if (!getAttribute(attributes, 'gutter')) {
+      attributes.push(
+        j.jsxAttribute(j.jsxIdentifier('edge'), j.literal('start'))
+      )
+    }
+
+    attributes.forEach(attribute => {
+      // replace error={true} by color='secondary'
+      if (
+        getAttributeName(attribute) === 'error' &&
+        getAttributeValue(attribute) === true
+      ) {
+        attributes.push(
+          j.jsxAttribute(j.jsxIdentifier('color'), j.literal('secondary'))
+        )
+        remove(attributes, attribute => getAttributeName(attribute) === 'error')
+      }
+
+      // replace gutter={true} by edge='start'
+      // remove gutter={false}
+      if (getAttributeName(attribute) === 'gutter') {
+        if (getAttributeValue(attribute) === true) {
+          attributes.push(
+            j.jsxAttribute(j.jsxIdentifier('edge'), j.literal('start'))
+          )
+        }
+        remove(
+          attributes,
+          attribute => getAttributeName(attribute) === 'gutter'
+        )
+      }
+    })
+  }
+
+  root.findJSXElements('Radio').forEach(transformRadio)
+  wrapRadioInFormControlLabel()
+
+  // print
+  return root.toSource()
```

Comment:
in my opinion, there are too many comments in this file. Does that comes from the web? If so, let's add the link.

If not, let's clean it. Each comment deserves, in my opinion, a well named method to explicit the behaviour. Example: 
```
const removeUselessAttributeFromRadio = () => {
      remove(attributes, attribute => getAttributeName(attribute) === 'value')
      remove(attributes, attribute => getAttributeName(attribute) === 'label')
}
```

Reply:
Non ça ne vient pas du net.

Je ne vois pas trop l'intérêt de faire des fonctions si elles ne sont utilisées qu'une seule fois. De plus certaines fonctions font plusieurs choses, ou selon certaines condition, ça me paraît compliquer de trouver un nom simple pour la décrire.

Enfin je ne suis pas convaincu que ce genre de refactor soit un "no go" pour une PR, si ? (le "no go" est d'ailleurs un outil qu'on utilise franchement très peu...)

Reply:
C'est du nit, mais c'est du clean code. Je suis satisfait déjà du simple test qui explique/documente/est garant du bon fonctionnement du code. Je ne vois plus l'intérêt du commentaire en haut de fichier, mais c'est toi qui vois. :-)